build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/biomeTasks.ts

```diff
@@ -0,0 +1,74 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import path from "path";
+import { GitRepo } from "../../../common/gitRepo";
+import { LeafWithFileStatDoneFileTask } from "./leafTask";
+import { getResolvedFluidRoot } from "../../../common/fluidUtils";
+import findUp from "find-up";
+
+export class BiomeTask extends LeafWithFileStatDoneFileTask {
+	private repoRoot: string | undefined;
+	private gitRepo: GitRepo | undefined;
+
+	/**
+	 * Includes all files in the task's package directory and any biome config files in the directory tree. Files ignored
+	 * by git are excluded.
+	 */
+	protected async getInputFiles(): Promise<string[]> {
+		this.repoRoot ??= await getResolvedFluidRoot(true);
+		this.gitRepo ??= new GitRepo(this.repoRoot);
+
+		const configFiles = await this.getBiomeConfigPaths(this.node.pkg.directory);
+		const files = (await this.gitRepo.getFiles(this.node.pkg.directory)).map((file) =>
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			path.join(this.repoRoot!, file),
+		);
+
+		return [...new Set([...configFiles, ...files])];
+	}
+
+	protected async getOutputFiles(): Promise<string[]> {
+		// Input and output files are the same.
+		return this.getInputFiles();
+	}
+
+	/**
+	 * Returns an array of all the biome config files found from the current working directory up to the root of the repo.
+	 *
+	 * Rather than parse and read the config files, this implementation naively searches for all config files from the
+	 * task's package directory up to the root of the repo and assumes they're all used. In the future we might want to
+	 * parse the config files anyway to extract ignore paths, at which point this implementation can change.
+	 */
+	private async getBiomeConfigPaths(cwd: string): Promise<string[]> {
+		this.repoRoot ??= await getResolvedFluidRoot(true);
+
+		const config = await findUp(["biome.json", "biome.jsonc"], { cwd });
+		if (config === undefined) {
+			if (isPathAbove(cwd, this.repoRoot)) {
+				// The version of find-up we're using (5.0) doesn't support a stop directory and upgrading is challenging since
+				// it's now ESM-only. This works around by stopping once we've stepped beyond the repo root.
+				return [];
+			}
```

Comment:
This logic does not appear to be correct based on the intent you described.
You want the config files within repo and none outside, but you only stop once you don't find config files and are out of repo instead of once out of the repo.
If there are any config files in directories above the repo root, current logic will pick them all up.
cc: @DLehenbauer

Reply:
I'll fix this in another pr. Just move the path above check earlier.

Reply:
Fixed in https://github.com/microsoft/FluidFramework/pull/20275

Reply:
I was just coming here to note the ESM issue. Here is a commit that moves to findUp 7.0 and also addresses the logic issue: https://github.com/jason-ha/FluidFramework/commit/d5add94776155cbcf099105421f8cab39d274cc7